arviz/data/inference_data.py

```diff
@@ -173,6 +175,53 @@ def __delattr__(self, group):
     def _groups_all(self):
         return self._groups + self._groups_warmup
 
+    def extend_xr_method(func):  # pylint: disable=no-self-argument
+        """Wrapper to add methods to InferenceData Class"""
+
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+
+            _filter = kwargs.pop("filter_groups", None)
+            _groups = kwargs.pop("groups", None)
+            _inplace = kwargs.pop("inplace", False)
+            self = args[0]
+            args = list(args)
```

Comment:
Can you give a small comment about the args order? What should be there?

The first args[0] is self, but also data in later steps?

Reply:
I have changed it.